projects/storefrontlib/cms-components/myaccount/new-consent-management/components/new-consent-management.component.ts

```diff
@@ -0,0 +1,340 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import {
+  AnonymousConsentsConfig,
+  AnonymousConsentsService,
+  AuthService,
+  ConsentTemplate,
+  GlobalMessageService,
+  GlobalMessageType,
+  UserConsentService,
+} from '@spartacus/core';
+import {
+  BehaviorSubject,
+  combineLatest,
+  concat,
+  Observable,
+  Subscription,
+} from 'rxjs';
+import {
+  distinctUntilChanged,
+  filter,
+  map,
+  scan,
+  skipWhile,
+  tap,
+  withLatestFrom,
+} from 'rxjs/operators';
+import { ConsentManagementComponentService } from '../../consent-management/consent-management-component.service';
+
+@Component({
+  selector: 'cx-new-consent-management',
+  templateUrl: './new-consent-management.component.html',
+})
+export class NewConsentManagementComponent implements OnInit, OnDestroy {
+  private subscriptions = new Subscription();
+  private allConsentsLoading = new BehaviorSubject<boolean>(false);
+
+  templateList$: Observable<ConsentTemplate[]>;
+  loading$: Observable<boolean>;
+
+  requiredConsents: string[] = [];
+
+  constructor(
+    protected userConsentService: UserConsentService,
+    protected globalMessageService: GlobalMessageService,
+    protected anonymousConsentsConfig: AnonymousConsentsConfig,
+    protected anonymousConsentsService: AnonymousConsentsService,
+    protected authService: AuthService,
+    protected consentManagementComponentService?: ConsentManagementComponentService
```

Comment:
Why is this an optional service?

Reply:
The new consent management component for cdc/cdp doesn't need much new logic, this file was copied from the existing consent-management.component.ts. I checked the commit history, this line was committed for cdc topic. see [consent-management.component.ts#L54]( https://github.com/SAP/spartacus/blame/develop-6.5.x/projects/storefrontlib/cms-components/myaccount/consent-management/components/consent-management.component.ts#L54). Could we keep this? To be honest, I have limited knowledge with cdc. Even through it tested well without `?` when connecting commerce ccv2 env p1, I am not sure if it will cause any error when integrated with cdc.